src/storage/LockingResourceStore.ts

```diff
@@ -55,4 +54,24 @@ export class LockingResourceStore implements AtomicResourceStore {
       await lock.release();
     }
   }
+
+  private async lockedRunGet(identifier: ResourceIdentifier, preferences: RepresentationPreferences,
```

Comment:
I have trouble parsing the name of the function. What does `lockedRunGet` mean?
We probably also want a comment on top explaining what it does.

Also, shall we
```suggestion
  protectded async lockedRunGet(identifier: ResourceIdentifier, preferences: RepresentationPreferences,
```
?

Reply:
Should the `lockedRun<T>` function also be changed to protected?

Reply:
👍